01_install_requirements.sh

```diff
@@ -16,122 +16,17 @@ if selinuxenabled ; then
-    else
-        sudo yum -y install epel-release --enablerepo=extras
-    fi
-fi
-
-# Install required packages
-# python-{requests,setuptools} required for tripleo-repos install
-if [ "${RHEL8}" = "True" ] ; then
-    sudo dnf install -y \
-        python36 \
-        python3-requests \
-        python3-setuptools
-    sudo alternatives --set python /usr/bin/python3
-
-    # TODO(russellb) - Install an rpm for this once OSP for RHEL8 is out
-    pushd ~
-    if [ ! -d crudini ] ; then
-        git clone https://github.com/pixelb/crudini
-    fi
-    pushd crudini
-    git pull -r
-    sudo pip3 install -U .
-    popd ; popd
-fi
-
-if [ "${RHEL8}" = "True" ] ; then
-    sudo subscription-manager repos --enable=ansible-2-for-rhel-8-x86_64-rpms
-
-    # make sure additional requirments are installed
-    sudo yum -y install \
-      ansible \
-      podman \
-      network-scripts \
-      ipmitool \
-      redhat-lsb-core
-
-    # TODO(russellb) - Install an rpm for this once OSP for RHEL8 is out
-    #sudo dnf groupinstall -y "Development Tools"
-    # workaround https://bugzilla.redhat.com/show_bug.cgi?id=1750866
-    sudo dnf groupinstall -y "Development Tools" --nobest
-    sudo dnf install -y python36-devel
-
-    # TODO(russellb) - Install an rpm for this once OSP for RHEL8 is out
-    pushd ~
-    if [ ! -d virtualbmc ] ; then
-        git clone https://git.openstack.org/openstack/virtualbmc
-    fi
-    pushd virtualbmc
-    git pull -r
-    sudo pip3 install -U .
-    curl 'https://review.rdoproject.org/r/gitweb?p=openstack/virtualbmc-distgit.git;a=blob_plain;f=virtualbmc.service;hb=HEAD' > virtualbmc.service
-    sed -i 's|/usr/bin/vbmcd|/usr/local/bin/vbmcd|' virtualbmc.service
-    sudo mv virtualbmc.service /etc/systemd/system/.
-    sudo systemctl daemon-reload
-    popd ; popd
-
-    pushd ~
-    if [ ! -d openstackclient ] ; then
-        git clone https://git.openstack.org/openstack/openstackclient
-    fi
-    pushd openstackclient
-    git pull -r
-    sudo pip3 install -U .
-    popd ; popd
-else
-    # We're reusing some tripleo pieces for this setup so clone them here
-    cd
-    if [ ! -d tripleo-repos ]; then
-      git clone https://git.openstack.org/openstack/tripleo-repos
-    fi
-    pushd tripleo-repos
-    sudo python setup.py install
-    popd
-
-    # Needed to get a recent python-virtualbmc package
-    sudo tripleo-repos current-tripleo
-
-    # There are some packages which are newer in the tripleo repos
-    sudo yum -y update
-
-    # make sure additional requirments are installed
-    sudo yum -y install \
-      ansible \
-      podman \
-      redhat-lsb-core
-fi
-
 export REPO_PATH=${WORKING_DIR}
-sync_repo_and_patch metal3-dev-env https://github.com/metal3-io/metal3-dev-env.git
-
-VM_SETUP_PATH="${REPO_PATH}/metal3-dev-env/vm-setup"
-
+sync_repo_and_patch metal3-dev-env https://github.com/metal3-io/metal3-dev-env.git https://patch-diff.githubusercontent.com/raw/metal3-io/metal3-dev-env/pull/54.patch
```

Comment:
It would be nice if we could just clone repos and not modify them if they already exist. It makes testing local changes way way easier.

Reply:
Yeah agreed, it actually caught me out when testing locally but I figured I'd leave the current behavior alone in this PR and maybe change it in a follow-up.

One issue is how to ensure the dev-scripts and metal3-dev-env repos don't get misaligned, since with this reuse of scripts they're fairly tightly coupled.  I guess we could try to pull --rebase any existing repo, and assume if the repo isn't on master the user knows what they're doing and is keeping it updated?  I guess that's similar to the go repos.

Reply:
What if we check the branch that metal3-dev-env is on and only update it if it's on master?

Reply:
> What if we check the branch that metal3-dev-env is on and only update it if it's on master?

Yes sounds good, and perhaps I'll also add a variable so you can use a metal3-dev-env from a non-default location (e.g we check it out to $WORKING_DIR atm, but I'd prefer my development version to live under my home directory).